build_barback/lib/src/transformer/transformer.dart

```diff
@@ -60,12 +60,11 @@ class BuilderTransformer implements Transformer, DeclaringTransformer {
       return;
     }
 
-    // Run the build step.
-    var logger = new Logger('${input.id}');
-    Logger.root.level = Level.ALL;
-    var logSubscription = logger.onRecord.listen((LogRecord log) {
-      if (log.loggerName != logger.fullName) return;
+    hierarchicalLoggingEnabled = true;
```

Comment:
Note that build_runner also messes with this setting.... in general I really hate how this setting can't be scoped somehow to a specific set of loggers :(